posts/solving-parallel-letter-frequency-elixir-part-2-adding-concurrency.md

```diff
@@ -0,0 +1,127 @@
+Exercises on Exercism are small, synthetic, and often seemingly trivial. It’s easy to imagine that experienced practitioners would have nothing to learn from them. However, solving these synthetic problems can push you to learn and apply parts of your language that you may not have explored. This new learning can lead you to solve real world problems more efficiently or in a more expressive way.
+
+[Parallel Letter Frequency](https://exercism.io/tracks/elixir/exercises/parallel-letter-frequency) is a medium difficulty exercise on [Exercism's Elixir Track](https://exercism.io/tracks/elixir) that unpacks a surprising number of interesting lessons. One of those lessons is how easy it is to add concurrency to Elixir code with core features alone, which has clear benefits for anyone writing applications whose performance could be improved by parallelizing tasks. 
+
+This exercise requires you to implement a `Frequency.frequency/2` function that determines letter frequency in a list of strings, with the added challenge of distributing the calculation to a number of worker processes, set with the `workers` argument:
+
+```elixir
+iex> Frequency.frequency(["Freude", "schöner", "Götterfunken"], workers)
+%{
+  "c" => 1, 
+  "d" => 1, 
+  "e" => 5, 
+  ...
+  "ö" => 2
+}
+```
+
+Let's explore how we can add concurrency to our solution using Elixir.
```

Comment:
Here you say _concurrency_ and below you say _parallelise_. These are somewhat related terms but don't mean the same thing, if I understand correctly.

Should we add some clarification about the difference?

Reply:
My intention was to use them synonymously, I'll read through the article again and see if I can change the wording to make it more clear.

Reply:
Thanks. Since they aren't synonyms I think it's important to be certain which this actually addresses.

My favorite talk on the subject is Rob Pike's "Concurrency is not parallelism" https://blog.golang.org/concurrency-is-not-parallelism